bikes/views.py

```diff
@@ -28,6 +34,47 @@ def create(self, request, *args, **kwargs):
         )
 
 
-class RentedBikesListAPIView(ListAPIView):
+class RentedBikesViewSet(CreateModelMixin, ListModelMixin, viewsets.GenericViewSet):
     queryset = Bike.objects.filter(status=BikeStatus.rented)
-    serializer_class = ReadBikeSerializer
+
+    def get_serializer_class(self):
+        if self.action == "create":
+            return RentBikeSerializer
+        else:
+            return ReadBikeSerializer
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        headers = self.get_success_headers(serializer.data)
+        try:
+            rented_bike = Bike.objects.get(id=request.data.get("id"))
+        except Bike.DoesNotExist:
+            return Response(
+                {"message": "bike not found"}, status=status.HTTP_404_NOT_FOUND
+            )
+        if rented_bike.user.username == "blocked":  # placeholder user is blocked
+            return Response(
+                {"message": "Blocked users are not allowed to rent bikes"},
+                status=status.HTTP_403_FORBIDDEN,
+            )
+        try:
+            Station.objects.get(id=rented_bike.station.id, state=StationState.blocked)
+        except Station.DoesNotExist:
+            if not rented_bike.status == BikeStatus.available:
+                return Response(
+                    {
+                        "message": "Bike is not available, it is rented, blocked or reserved"
+                    },
+                    status=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                )
+            rented_bike.status = BikeStatus.rented
```

Comment:
Sama zmiana pola obiektu nie zapisuje tego do bazy (modyfikujemy tylko read reprezentację obiektu w bazie). Trzeba zawsze pamiętać o 
```
rented_bike.save()
```

Dodatkowo, wypożyczony rower przestaje być przypisany do stacji, trzeba pole `rented_bike.station` ustawić na `None`.

Ostatecznie powinno to wyglądać tak:
```
rented_bike.status = BikeStatus.rented
rented_bike.station = None
rented_bike.save()
```

Reply:
Dodatkowa sprawa, zabrakło testów które mogłyby to wyłapać, dodaj proszę takie dwa testy:
```
    def test_rent_bike_status_changes_to_rented(self):

    def test_rent_bike_station_changes_to_null(self):
```